packages/next/src/server-build.ts

```diff
@@ -1131,6 +1131,23 @@ export async function serverBuild({
     });
   }
 
+  if (appDir) {
+    // create .rsc variant for app lambdas and edge functions
+    // to match prerenders so we can route the same when the
+    // __flight__ header is present
+    Object.values(appPathRoutesManifest || {}).forEach(route => {
```

Comment:
Not sure if performance is a concern, but a for-of loop would be more efficient as V8 wouldn't have to allocate a stack frame for each iteration.

Reply:
Updated in https://github.com/vercel/vercel/pull/8611/commits/81ad6360e2bb6afa8cb05cd96aa80095bd906420